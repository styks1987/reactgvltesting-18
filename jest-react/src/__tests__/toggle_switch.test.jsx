/** @format */

import React from 'react';
// Simple way to render components
import renderer from 'react-test-renderer';
// We need a more complicated way to render our components
import {mount, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({adapter: new Adapter()});
// Our component
import ToggleSwitch from '../toggle_switch';

// Just do a simple snapshot test
test('<ToggleSwitch /> renders with props on', () => {
    const component = renderer.create(<ToggleSwitch on={true} />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

// Test and assert by simulating click events and comparing button text
test('<ToggleSwitch /> can be toggled', () => {
    const component = mount(<ToggleSwitch />);

    expect(component.find('.btn-toggle').text()).toEqual('Turn Me ON');

    component.find('.btn-toggle').simulate('click');

    expect(component.find('.btn-toggle').text()).toEqual('Turn Me OFF');
});
