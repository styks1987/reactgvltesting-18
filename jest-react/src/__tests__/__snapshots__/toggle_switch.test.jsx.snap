// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleSwitch /> renders with props on 1`] = `
<div
  style={
    Object {
      "background": "green",
      "color": "white",
      "fontSize": "30px",
      "fontWeight": "bold",
      "padding": "50px",
    }
  }
>
  I AM ON
   
  <button
    className="btn-toggle"
    onClick={[Function]}
  >
    Turn Me OFF
  </button>
</div>
`;
